getnacelle/nacelle-js
# feat(shopify-cart): Adds params for setting country and language codes

## WHY are these changes introduced?

Addresses [ENG-7060](https://nacelle.atlassian.net/browse/ENG-7060) - Adds parameters to `cartCreate` for setting the `language` and `country` in all Cart queries and mutations. This allows the cart to support internationalized stores. 



## WHAT is this pull request doing?

1. Adds `language` and `country` params to the `createCart` function that correspond to the Shopify GQL [Language Code][shopify-cart-language-code] and [Country Code][shopify-cart-country-code] enums. If not provided, the language code defaults to `EN` and the country code corresponds to `ZZ`. Note that prior to this PR, no language code was used and all queries used `ZZ` for the country code.
2. Updates All Shopify Enums to be types - this makes them easily accessible as strings, which makes a better public API for users. 
3. Updates every query/mutation GQL to use accept a `language` variable parameter, and to use the `language` in the `inContext` directive.
4. Passes the `language` & `country` variables to every action, and uses them in the queries. Updates the types passed to the `GQLClient` in each action to use the `<QueryName>Variables` type so that `language` and `country`  are expected. This should allow us to update query variables without needing to change types in the future too. 
5. Adds `language` and `country` to the args/responses for all tests.
6. Adds a test for passing a non-default language to `createCart` & a test passing a non-default `country` to `createCart`
7. Updates the `README` to document the 2 new parameters to `createCart`. 


## How to Test

1. Clone this branch `mdarrik/shopify-cart/ENG-7060-add-params-for-setting-and-language` (or `gh pr checkout #` if using the gh cli)
2. Run `npm run test:ci` - all tests should passed
3. Run `npm run build` - build should complete successfully
4. Use `npm link` to test a local project featuring shopify cart. Outgoing queries should include the `language` and `country` specified, or default to `EN` and `ZZ` respectively.



[shopify-cart-language-code]: https://shopify.dev/api/storefront/2022-07/enums/LanguageCode
[shopify-cart-country-code]: https://shopify.dev/api/storefront/2022-07/enums/CountryCode
